pkgs/development/compilers/ghc/8.2.2.nix

```diff
@@ -70,11 +70,9 @@ let
     ++ stdenv.lib.optional (!enableIntegerSimple) gmp
     ++ stdenv.lib.optional (platform.libc != "glibc") libiconv;
 
-  toolsForTarget =
-    if hostPlatform == buildPlatform then
-      [ targetPackages.stdenv.cc ] ++ stdenv.lib.optional useLLVM llvmPackages.llvm
-    else assert targetPlatform == hostPlatform; # build != host == target
-      [ stdenv.cc ] ++ stdenv.lib.optional useLLVM buildLlvmPackages.llvm;
+  toolsForTarget = [
+    pkgsBuildTarget.targetPackages.stdenv.cc
```

Comment:
Why do you need targetPackages here?

Reply:
because `stdenv.cc` comes from the previous `stage. `targetPackages.stdenv` sort of cancels out so I get something like `pkgsBuildTarget.gcc` or `pkgsBuildTarget.llvm` in the end.